spring amqp 2.1.9 has an invalid POM

I am trying to consume the Spring AMQP `2.1.9.RELEASE` to release the next maintenance release of Spring Boot.

It looks like the published pom is invalid:

```
[WARNING] The POM for org.springframework.amqp:spring-amqp:jar:2.1.9.RELEASE is invalid, transitive dependencies (if any) will not be available: 3 problems were encountered while building the effective model for org.springframework.amqp:spring-amqp:2.1.9.RELEASE
[ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.core:jackson-core:jar is missing. @
[ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.core:jackson-databind:jar is missing. @
[ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.dataformat:jackson-dataformat-xml:jar is missing. @
```

I can see that [the released pom](https://search.maven.org/artifact/org.springframework.amqp/spring-amqp/2.1.9.RELEASE/jar) has indeed a reference to 3 jackson artifacts without a version.

--- a/publish-maven.gradle
+++ b/publish-maven.gradle
@@ -14,6 +14,16 @@ install {
 
 def customizePom(pom, gradleProject) {
 	pom.whenConfigured { generatedPom ->
+
+		// sort to make pom dependencies order consistent to ease comparison of older poms
+		generatedPom.dependencies = generatedPom.dependencies.sort { dep ->
+			"$dep.scope:$dep.groupId:$dep.artifactId"
+		}
+
+		def managedVersions = dependencyManagement.managedVersions
+		generatedPom.dependencies.findAll{dep -> !dep.version }.each { dep ->
+			dep.version = managedVersions["${dep.groupId}:${dep.artifactId}"]
+		}
 		// respect 'optional' and 'provided' dependencies
 		gradleProject.optionalDeps.each { dep ->
 			generatedPom.dependencies.find { it.artifactId == dep.name }?.optional = true

